src/PolyType.SourceGenerator/Parser/Parser.cs

```diff
@@ -20,31 +20,96 @@ public sealed partial class Parser : TypeDataModelGenerator
         CommonHelpers.CreateTupleComparer<ITypeSymbol, string>(
             SymbolEqualityComparer.Default,
             CommonHelpers.CamelCaseInvariantComparer.Instance);
-    
+
     private readonly PolyTypeKnownSymbols _knownSymbols;
+    private readonly ImmutableDictionary<INamedTypeSymbol, TypeExtensionModel> _typeShapeExtensions;
```

Comment:
Do we need to use ImmutableDictionary? It's not the fastest collection and I presume we're not doing persistable updates either given that the field is readonly.

Reply:
I'll change it to `IReadOnlyDictionary`